aws/retry/middleware.go

```diff
@@ -64,19 +64,15 @@ func (r Attempt) logf(logger logging.Logger, classification logging.Classificati
 func (r Attempt) HandleFinalize(ctx context.Context, in smithymiddle.FinalizeInput, next smithymiddle.FinalizeHandler) (
 	out smithymiddle.FinalizeOutput, metadata smithymiddle.Metadata, err error,
 ) {
-	var attemptNum, retryCount int
+	var attemptNum int
 	var attemptClockSkew time.Duration
+	var shouldRetry bool
```

Comment:
Looks like `shouldRetry` is only used in side of the for lop.  It probably should be moved into that scope.  Is there a reason it should be persisted between loops?